File: CodeForces/301-400/381/A/381A.r
# Problem 381A: Sereja and Dima
# https://codeforces.com/contest/381/problem/A
# Executed with: Rscript (R) version 4.5.1 (2025-06-13)
# Linux version: Linux kernel 6.17.0-6-generic
# Written on: October 28th, 2025
# Codeforces language used: Cannot be submitted

options(scipen=999)

w <- as.integer(scan(file = "stdin", nmax = 1, quiet = TRUE))
nums <- scan(file = "stdin", nmax = w, quiet = TRUE)

turn <- 0
sereja_score <- 0
dima_score <- 0

while (w > turn) {
    if (nums[1] > nums[length(nums)]) {
        if (turn %% 2 == 0) {
            sereja_score <- sereja_score + nums[1]
        } else {
            dima_score <- dima_score + nums[1]
        }
        nums <- nums[-1]
    } else {
        if (turn %% 2 == 0) {
            sereja_score <- sereja_score + nums[length(nums)]
        } else {
            dima_score <- dima_score + nums[length(nums)]
        }
        nums <- nums[-length(nums)]
    }

    turn <- turn + 1
}
cat(sereja_score, dima_score, "\n")

File: CodeForces/301-400/381/A/381A.cs
// Problem 381A: Sereja and Dima
// https://codeforces.com/contest/381/problem/A
// Compiled with: Mono JIT compiler version 6.12.0.199 (tarball Fri May  2 12:22:51 UTC 2025)
// Linux version: Linux kernel 6.17.0-6-generic
// Submitted on: November 1st, 2025
// Codeforces language used: C# Mono 6.8

using System;
using System.Collections.Generic;
using System.Linq;

public class Scanner {
    private Queue<string> buffer = new Queue<string>();

    private void Refill() {
        string line = null;
        do {
            line = Console.ReadLine();
            if (line == null) return;
            line = line.Trim();
        } while (line.Length == 0);

        foreach (var token in line.Split((char[])null, StringSplitOptions.RemoveEmptyEntries))
            buffer.Enqueue(token);
    }

    public int nextInt() {
        if (buffer.Count == 0) Refill();
        return int.Parse(buffer.Dequeue());
    }

    public double nextDouble() {
        if (buffer.Count == 0) Refill();
        return double.Parse(buffer.Dequeue(), System.Globalization.CultureInfo.InvariantCulture);
    }

    public long nextLong() {
        if (buffer.Count == 0) Refill();
        return long.Parse(buffer.Dequeue());
    }

    public string nextWord() { 
        if (buffer.Count == 0) Refill(); 
        return buffer.Dequeue(); 
    }

    public string nextString() {
        if (buffer.Count == 0) Refill(); 
        string val = string.Join(" ", buffer);
        buffer.Clear();
        return val;
    }

    public List<int> nextIntArray(bool sorted = false) {
        if (buffer.Count == 0) Refill();
        var arr = buffer.Select(int.Parse).ToList();
        buffer.Clear();
        if (sorted) arr.Sort();
        return arr;
    }

    public List<long> nextLongArray(bool sorted = false) {
        if (buffer.Count == 0) Refill();
        var arr = buffer.Select(long.Parse).ToList();
        buffer.Clear();
        if (sorted) arr.Sort();
        return arr;
    }
}

public class Program {
    public static void Main() {
        Scanner scanner = new Scanner();
        int amount_of_nums = scanner.nextInt();
        List<int> all_nums = scanner.nextIntArray();

        int sereja_score = 0;
        int dima_score = 0;

        for (int i = 0; i < amount_of_nums; i++) {
            int index = -1;
            if (all_nums[0] > all_nums[all_nums.Count()-1]) {
                index = 0;
            } else {
                index = all_nums.Count()-1;
            }

            if (i % 2 == 0) {
                sereja_score += all_nums[index];
            } else {
                dima_score += all_nums[index];
            }
            all_nums.RemoveAt(index);
        }

        Console.WriteLine($"{sereja_score} {dima_score}");
    }
}
